### ISSUE TITLE
SCOPE support: prefill-only trajectory scoring (Model.score_trajectories)

### ISSUE THREAD

**Issue by mmcdermott** [mmcdermott/MEDS_EIC_AR#95]
## Summary

Add a `Model.score_trajectories(prompt_ids, generated_ids, target_token_ids)` method that runs a single prefill-only forward pass over `cat([prompt, generated])` per trajectory and returns per-position target-event probabilities. This is the generation-side infrastructure for SCOPE (M1 estimator from [arXiv:2602.03730](https://arxiv.org/abs/2602.03730)).

## What SCOPE needs from this repo

SCOPE computes `sum_t P(target_event | history_<t)` across decoding positions. The reference implementation does this as a **separate prefill-only scoring pass** after generation completes — generation is standard autoregressive sampling, then the full `(prompt + generated)` sequence is fed back through the model in one forward pass to extract per-position logprobs for the target event token(s).

On HF this is ~10 lines (one `model.forward()` call + softmax + gather). On SGLang/vLLM it's a native prefill-with-logprobs API call. The key requirement is that **this must benefit from prefix caching** — when scoring N trajectories that share the same subject prompt, the KV cache for the shared prefix should be computed once, not N times.

## Implementation sketch

```python
def score_trajectories(
    self,
    prompt_ids: torch.Tensor,       # [B, L_prompt]
    generated_ids: torch.Tensor,     # [B, L_gen]
    target_token_ids: Sequence[int], # which tokens to extract p(target) for
) -> torch.Tensor:                   # [B, L_gen, len(target_token_ids)]
    full_seq = torch.cat([prompt_ids, generated_ids], dim=1)
    attn_mask = full_seq != self.PAD_INDEX
    outputs = self.HF_model(input_ids=full_seq, attention_mask=attn_mask)
    logits = outputs.logits[:, prompt_ids.size(1) - 1 : -1]  # shifted to align with generated positions
    probs = F.softmax(logits, dim=-1)
    target_probs = probs[..., target_token_ids]
    return target_probs
```

## Output schema extension

`GeneratedTrajectorySchema` should optionally carry per-position target-event probability so `MEDS_trajectory_evaluation` can compute the SCOPE estimator without re-scoring.

## Dependencies

- Independent of generation path changes (#82, #86).
- Benefits hugely from prefix caching → #88 (SGLang) or vLLM.
- Benefits from N-trajectories interleaving → #89.
- The estimator math (aggregation, confidence intervals) lives in `MEDS_trajectory_evaluation`, not here.
- Shared interface with REACH: both need `target_token_ids` lookup (string code → vocab index).

## References

- [arXiv:2602.03730](https://arxiv.org/abs/2602.03730) — SCOPE/REACH paper
- [SCOPE_REACH_optimized_inference](https://github.com/lukesolo-ml/SCOPE_REACH_optimized_inference) — reference implementation on SGLang
